staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go

```diff
@@ -53,7 +53,7 @@ type Object interface {
 	SetDeletionGracePeriodSeconds(*int64)
 	GetLabels() map[string]string
 	SetLabels(labels map[string]string)
-	GetAnnotations() map[string]string
+	GetAnnotations() (map[string]string, error)
```

Comment:
@smarterclayton thoughts on making these capable of returning errors? I'd want to do it consistently for all the methods, not piece-mill like this, but that's a huge change

Reply:
This isn't the right place to add this.  What you really want is a variant of Accessor https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go#L102 that returns an error if you're not a metadata object.  Determining that you're a metadata object is a duck check for unstructured content, but you should be able to do that before you get here.

Reply:
An object that is a metadata.Object is required by law to support annotations, so it would be inappropriate to have an optional error on that interface.  You just shouldn't be able to get an Object for whatever type you need.